src/searchbar.cpp

```diff
@@ -97,6 +44,52 @@ SearchBar::SearchBar(QWidget *parent) :
     connect(this, &QLineEdit::returnPressed, this, [=]() {
         m_returnPressed = true;
     });
+
+    // placing an icon inside QLineEdit
+    addAction(m_action, QLineEdit::LeadingPosition);
+    setSearchMode();
+    connect(m_action, &QAction::triggered,
+            this, &SearchBar::on_action_triggered);
+}
+
+
+void SearchBar::setSearchMode()
+{
+    clear();    // prepare for entering search request
+    setReadOnly(false);
+
+    m_action->setEnabled(false);    // ignore clicking on 'Search' icon
+
+    m_action->setIcon(QIcon(":/icons/search.svg"));
+
+    /* Enable blinking cursor in QLineEdit
+     *
+     * There is no method to do it: QLineEdit doens't have API for controlling
+     * showing of text cursor.
+     *
+     * This workaround was found in QLineEdit sources:
+     * d->control->setBlinkingCursorEnabled(true)
+     * is called only from two places:
+     *  - from focusInEvent() - but we have to ignore it: at the time the event arrived
+     *    it could be action toggling next;
+     *  - from event() if (e->type() == QEvent::Show) - we use it here:
+     */
+    QApplication::postEvent(this, new QShowEvent());
```

Comment:
This look like a issue for Qt
When the blinking of the cursor is disable ?
Should it be enable "itself" ? If not QLineEdit need a new method.

We will still need the workaround here (if we have properly used Qt) but it would be nice to create a issue on Qt side.

Reply:
I don't think it's Qt bug. We just want our own custom behaviour.

It somes from Apple Safari:

When there is no opened page, we see 'Search' icon in the address bar:

![Screenshot from 2022-03-05 01-02-41](https://user-images.githubusercontent.com/5078434/156818162-f1bbe68b-edee-4701-95d9-505f78f60fc3.png)

When some page is open, we see 'Add to reading list' button:

![Screenshot from 2022-03-05 01-04-17](https://user-images.githubusercontent.com/5078434/156818281-719fdfb1-c0b1-4bc1-8965-26f31d4cf867.png)

But once we start typing, it turns into 'Search' icon again:

![Screenshot from 2022-03-05 01-10-57](https://user-images.githubusercontent.com/5078434/156818830-bd2ac2fe-c0da-43ef-870f-b99756bd98e3.png)

So this is why we want only one action, not two.